### PR TITLE
Bolt Build Tools: v0.1.0-beta.1

### PR DESCRIPTION
**Pull request by sghoweri**
Initial stable pre-release of the front-end build tools that power Bolt. Additional cleanup, linting, and documentation are still in progress but this initial early release can fully build and serve fully working (and mostly fully featured) frontend environments in both Pattern Lab and Jekyll.

## Includes the following features
- Styles build tasks for compiling, cleaning, watching, linting, and autoprefixing your Sass
- Pattern Lab build task w/ live reload integration
- Lerna Auto-reload
- Pattern Lab <—> Packages symlink task
- Slack deployment task
- A BrowserSync-enabled HTTP2 local dev server w/ live reloading and automatic SSL certificate generation
- Bolt Build Core: shared build-related utilities and helper functions

## Additional features
- Eslint code quality coverage
- Babel code transpiling of build tasks
- Brand new ES6 gulp task architecture
- Lerna versioned components
- Internal upgrade to Gulp4
- Composable build tasks w/ preconfigured help text; allows for customizing and creating aliases to tasks on the fly
- Repackaged collection of all available build tools to simplify the onboarding process
- Serverside patternlab builds via Travis
- Travis CI configuration
- Very first round of frontend unit tests using Chai and Mocha (tests if the CSS build task is working as expected)
- Travis CI build integration to protect against breaking builds
- Slack notifications on unsuccessful builds
- Custom Slack notification on successful deployment to Surge.sh (WIP)
- Customizable build tasks via passed in parameters

CC @christophersmith262 @semiuniversal

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -22,6 +22,10 @@ public
 stuff
```
Comment (semiuniversal) [inline .gitignore]: Everyone needs a "stuff" folder ;)

File: yarn-error.log [hunk 2]
```diff
@@ -0,0 +1,8614 @@
+Arguments:
```
Comment (semiuniversal) [inline yarn-error.log]: Do we need to commit log files?

File: packages/elements-headings/src/scss/index.scss [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+
+.o-grid {
+  background: red;
+}
+
+
+
+
+// .c-button__icon {
```
Comment (semiuniversal) [inline packages/elements-headings/src/scss/index.scss]: Why commit new code that's commented out?

**Comment by sghoweri** [comment]
@mikemai2awesome @theSadowski closing this out -- latest code merged in earlier supersedes this

**Pull request closed without merging**